src/traverse/mod.rs

```diff
@@ -450,69 +451,135 @@ pub(crate) fn extract(
-                        // path.
-                        // Potentially become mdbook aware
-                        // <https://github.com/drahnr/cargo-spellcheck/issues/273>
-                        log::debug!(
-                            "Unknown file type encountered, skipping path: {}",
-                            path.display()
-                        );
-                        continue;
+
+    // stage 2 - check for manifest, .rs , .md files and directories
+    if gitignore {
+        let max_depth = match recurse {
+            true => None,
+            false => Some(1),
+        };
+
+        let mut all_files = VecDeque::<PathBuf>::with_capacity(32);
+
+        while let Some(path) = flow.pop_front() {
+            let files = ignore::WalkBuilder::new(path)
+                .git_ignore(true)
+                .max_depth(max_depth)
+                .same_file_system(true)
+                .skip_stdout(true)
+                .build()
+                .filter_map(|entry| {
+                    entry
+                        .ok()
+                        .filter(|entry| entry.file_type().map(|ft| ft.is_file()).unwrap_or(false))
+                        .map(|x| x.path().to_owned())
+                })
+                .filter(|path: &PathBuf| {
+                    path.to_str()
+                        .map(|x| x.to_owned())
+                        .filter(|path| {
+                            path.ends_with(".rs")
+                                || path.ends_with(".md")
+                                || path.ends_with("Cargo.toml")
+                        })
+                        .is_some()
+                });
+
+            all_files.extend(files);
+        }
+
+        while let Some(path) = all_files.pop_front() {
+            let x = if let Ok(meta) = path.metadata() {
+                if meta.is_file() {
+                    match path.file_name().and_then(|x| x.to_str()) {
+                        Some(file_name) if file_name == "Cargo.toml" => Extraction::Manifest(path),
+                        Some(file_name) if file_name.ends_with(".md") => Extraction::Markdown(path),
+                        Some(file_name) if file_name.ends_with(".rs") => Extraction::Source(path),
+                        _ => {
+                            // This branch is commonly entered when ran on a non-cargo
+                            // path.
+                            // Potentially become mdbook aware
+                            // <https://github.com/drahnr/cargo-spellcheck/issues/273>
+                            log::debug!(
+                                "Unknown file type encountered, skipping path: {}",
+                                path.display()
+                            );
+                            continue;
+                        }
                     }
+                } else {
+                    Extraction::Missing(path)
                 }
```

Comment:
I think this part should become a function, it's (almost?) identical to the one in the `else` branch.